### ISSUE TITLE
Door colors on the map.

### ISSUE THREAD

**Issue by dragon3025** [AM2R-Community-Developers/AM2R-Community-Updates#104]
It would be very convenient to have door colors on the map. It can be easy to forget a door type in a previous location.

**Comment by Miepee** [comment]
Thanks for opening the feature request!
Unfortunately, we don't have any plans to implement this, as we think it'd unnecessary clutter the minimap and make it less readable.

**Comment by dragon3025** [comment]
I don't expect this reply to change anyone's mind, it's just my 2 cents.

I don't see how it clutters since you're only recoloring what's already there and not adding anything new. Also, they've been coloring doors on the map since Metroid Fusion, and I've never heard anyone complain that the doors make it harder to read; it only adds convenience.

**Comment by ShirtyScarab554** [comment]
Clutter probably isn't the most accurate term to use there, but it would be kind of confusing / strange to have doors on the map given there aren't many doors in AM2R overall, and most other 2D titles didn't really have them properly until Dread introduced much more detailed maps.

Fusion's doors that were shown on the map were specifically the locked level doors, and those needed to be shown since they all had to be unlocked by a control panel which released them simultaneously in order to progress through the game.
(with how much the game points the player where to go though I think most people didn't pay much attention to the map icons)
I have seen a few people complain about the door icons in Fusion, though admittedly they blend into the rest of the map so much due to all of the pink that they're hard to notice for most of the people I've talked to.

Now with that all being said, from a technical standpoint it would be a LOT of work to implement something like this with how stuff is designed in AM2R currently.
It's not nearly as simple as just "Recolor what's already there and not add anything new.", we'd have to make new map icons for each door type (which would probably be the only easy thing to do), but then we'd have to hook those up as their own individual map icons and set them up on the map in the specific spots they would be located, and to my knowledge editing the map is already quite a hassle to do with how it was designed.
On top of that we'd have to try and utilize the games event system for all of the special doors in a way that when they're destroyed, the minimap will update to show blue doors instead of the special door colors, which would require its own set of changes in order to work with the minimap correctly.

TLDR; While it could be cool, it may or may not look strange and it would be a LOT of work to change the necessary systems to include a small visual feature.

**Comment by Zackonark** [comment]
tl;dr It's about the amount of effort it would take to include the feature versus how much it would add to the game. Adding door tiles to the map would not add very much because there are very few doors in the game, and like Shirty explained, they would be a lot harder to add than it sounds.
